dangerzone/gui/updater.py

```diff
@@ -0,0 +1,225 @@
+        """Check the new settings value with the old one, and optionally save it."""
+        old_val = self.dangerzone.settings.updater_get(key)
+        if val == old_val:
+            return
+
+        self.dangerzone.settings.updater_set(key, val)
+        self.dangerzone.settings.save()
+
+    @property
+    def first_run(self) -> bool:
+        return self.dangerzone.settings.updater_get("first_run")
+
+    @first_run.setter
+    def first_run(self, val: bool) -> None:
+        self._save_me_maybe("first_run", val)
+
+    @property
+    def check(self) -> Optional[bool]:
+        return self.dangerzone.settings.updater_get("check")
+
+    @check.setter
+    def check(self, val: bool) -> None:
+        self._save_me_maybe("check", val)
+
+    def prompt_for_checks(self) -> bool:
+        """Ask the user if they want to be informed about Dangerzone updates."""
+        log.debug("Prompting the user for update checks")
+        # TODO: Render HTML links properly
+        # FIXME: Handle the case where a user clicks on "X", instead of explicity making
+        # a choice. We should probably ask them again on the next run.
+        check = Alert(
+            self.dangerzone,
+            message=MSG_CONFIRM_UPDATE_CHECKS,
+            ok_text="Yes",
+            cancel_text="No",
+        ).launch()
+        return bool(check)
+
+    def should_check_for_updates(self) -> bool:
+        """Determine if we can check for updates based on settings and user prefs."""
+        log.debug("Checking platform type")
+        # TODO: Disable updates for Homebrew installations.
+        # TODO: Add a flag that will force-check for updates, so that we can test it on
+        # Linux. Once we have this flag, uncomment the following lines.
+        # if platform.system() == "Linux" and os.environ():
+        #     log.debug("Running on Linux, disabling updates")
+        #     self.check = False
+        #     return False
+
+        log.debug("Checking if first run of Dangerzone")
+        if self.first_run:
+            log.debug("Dangerzone is running for the first time, updates are stalled")
+            self.first_run = False
+            return False
+
+        log.debug("Checking if user has already expressed their preference")
+        if self.check is None:
+            log.debug("User has not been asked yet for update checks")
+            self.check = self.prompt_for_checks()
+            # TODO: Should we always check for updates in the third run, or can we check
+            # them immediately, if the user wants to enable them?
+            return False
+        elif not self.check:
+            log.debug("User has expressed that they don't want to check for updates")
+            return False
+
+        return True
+
+    def can_update(self, cur_version: str, latest_version: str) -> bool:
+        if cur_version == latest_version:
+            return False
+        elif cur_version > latest_version:
```

Comment:
Yeap, I'm aware of this limitation, thanks for flagging it. As you correctly identified, this will bite us only when we start having a `10` somewhere in the version number. I figured that by the time this happens, we will probably have a different implementation for update checks, maybe something close to an update framework.

Anw, it's not worth adding this artificial limitation to avoid an extra dependency. I'll add a fix for that.

Reply:
Should be fixed in 3fcf6f1.

Reply:
I think we're still missing in one other place:

https://github.com/freedomofpress/dangerzone/pull/466/files#diff-06aa7bd0d159b022df5598aad34c689a3133689d668097b11337cd675cd90a3bR72